File: android/src/main/kotlin/com/example/design_engine/layer2_core/dark/ETertiaryDark.kt
package com.example.design_engine.layer2_core.dark
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object ETertiaryDark {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyd33
            AppTheme.STANDARD -> R.color.cstd33
            AppTheme.BLUE -> R.color.cbd33
            AppTheme.GREEN -> R.color.cgd33
            AppTheme.RED -> R.color.crd33
            AppTheme.NOTHING -> R.color.cnod33
        }
    }
}

File: lib/layer2_core/dark/e_tertiary_dark.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/dark/system.dart';
import '../../layer1_raw/dark/standard.dart';
import '../../layer1_raw/dark/blue.dart';
import '../../layer1_raw/dark/green.dart';
import '../../layer1_raw/dark/red.dart';
import '../../layer1_raw/dark/nothing.dart';

class ETertiaryDark {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyd33;
      case 'standard': return cstd33;
      case 'blue': return cbd33;
      case 'green': return cgd33;
      case 'red': return crd33;
      case 'nothing': return cnod33;
      default: return cstd33;
    }
  }
}
